mod/workshop/locallib.php

```diff
@@ -3447,122 +3359,161 @@ protected function aggregate_grading_grades_process(array $assessments, $timegra
     }
 
     /**
-     * Returns SQL to fetch all enrolled users with the given capability in the current workshop
-     *
-     * The returned array consists of string $sql and the $params array. Note that the $sql can be
-     * empty if a grouping is selected and it has no groups.
+     * Gets users with any of the specified capabilities on the workshop activity.
      *
      * The list is automatically restricted according to any availability restrictions
      * that apply to user lists (e.g. group, grouping restrictions).
      *
-     * @param string $capability the name of the capability
-     * @param bool $musthavesubmission ff true, return only users who have already submitted
+     * @param array $capabilities array of capability names (If the user has ANY of them, it returns true for that user)
+     * @param bool $musthavesubmission if true, return only users who have already submitted
      * @param int $groupid 0 means ignore groups, any other value limits the result by group id
-     * @return array of (string)sql, (array)params
+     * @param int $limitfrom return a subset of records, starting at this point (optional, required if $limitnum is set)
+     * @param int $limitnum return a subset containing this number of records (optional, required if $limitfrom is set)
+     * @return array of users
      */
-    protected function get_users_with_capability_sql($capability, $musthavesubmission, $groupid) {
-        global $CFG;
-        /** @var int static counter used to generate unique parameter holders */
-        static $inc = 0;
-        $inc++;
-
-        // If the caller requests all groups and we are using a selected grouping,
-        // recursively call this function for each group in the grouping (this is
-        // needed because get_enrolled_sql only supports a single group).
-        if (empty($groupid) and $this->cm->groupingid) {
-            $groupingid = $this->cm->groupingid;
-            $groupinggroupids = array_keys(groups_get_all_groups($this->cm->course, 0, $this->cm->groupingid, 'g.id'));
-            $sql = array();
-            $params = array();
-            foreach ($groupinggroupids as $groupinggroupid) {
-                if ($groupinggroupid > 0) { // just in case in order not to fall into the endless loop
```

Comment:
Capitalise first char and full stop at end.
https://moodledev.io/general/development/policies/codingstyle#inline-comments

Reply:
This is also a removed section